### ISSUE TITLE
INTERVAL `x` SECOND values with `x<0.001` are silently truncated to zero

### ISSUE THREAD

**Issue by alec-heif** [trinodb/trino#6754]
minimal example:
```
presto:default> SELECT val + INTERVAL '.0001' second FROM (VALUES TIMESTAMP '2021-01-01 00:00:00.123456789 UTC') AS t (val);
               _col0
-----------------------------------
 2021-01-01 00:00:00.123456789 UTC
(1 row)
```

This is a really unfortunate bug as it results in silently incorrect data. Ideally the `INTERVAL` function could be patched to do the right thing here, but we'd also be ok if it would at least throw an error rather than return incorrect data.

**Comment by martint** [comment]
Good catch. Intervals currently are limited to precision = 3 (i.e., milliseconds), so it seems reasonable to fail if a literal with higher precision is used.

**Comment by alec-heif** [comment]
@martint just wanted to bump this issue since another of our users hit this today.

**Comment by mohitsaxenaknoldus** [comment]
Please assign it to me.

**Comment by mwesolowski** [comment]
@martint @findepi Could you take a look at the PR? Unfortunately I cannot assign reviewers.

**Comment by Por-Favor** [comment]
Hi! I'm a student in the CodePath AI301 open-source program, and I'd like to work on this issue. 

I read through the history: PR #13426 implemented the "fail on >3 fractional digits" behavior @martint described above.

My plan is to reproduce the bug locally this week, then pick up where that previous PR left off.

Just a  quick question:
-  Is failing with an error on INTERVAL literals exceeding millisecond precision still the desired behavior?

**Comment by dgvj-work** [comment]
I'd like to take this. Prior PR #13426 was closed without merging, and there hasn't been follow-up activity since. I'll reproduce on current main, implement failing on INTERVAL second literals with more than millisecond precision (as @martint suggested), add a test, and open a PR.
